### ISSUE TITLE
508-defect-3 [SCREENREADER] Individual disabilities currently in individual <dl> would be improved in a single <dl>

### ISSUE THREAD

**Issue by jenstrickland** [department-of-veterans-affairs/va.gov-team#4637]
# [SCREENREADER]: Individual disabilities currently in individual `<dl>` would be improved in a single `<dl>`

## Description

The individual disabilities are currently coded into individual `<dl>`, and the screenreader reads out as a definition list, then the term, then the individual data. When it gets to the next one, it does the same. It becomes a lot of repetition of the frame. Since it is a list of terms and their descriptive content, it would be improved as a single `<dl>` with the `<dt>` and `<dd>` chained within. This would improve the screenreader user's experience, by reducing the repetitive information that may distract or annoy.

## Point of Contact

Jennifer

## Acceptance Criteria

As a screenreader user, I want to understand the context and relationships of the content, with minimal extra noise.

## Environment

* Screenreading device: JAWS, NVDA
* Server destination: staging

## Steps to Recreate

1. Enter `https://staging.va.gov/disability/view-disability-rating/rating` in browser
2. Start screenreading device 
3. Navigate to the content
4. Verify that for each "card" the "framing" of a description list is read

### Current code example
```html
<dl class="vads-u-display--block vads-l-col--12 vads-u-background-color--gray-lightest vads-u-margin-x--0 vads-u-margin-top--0 vads-u-margin-bottom--2 vads-u-padding-top--1 vads-u-padding-bottom--2 vads-u-padding-x--2">
  <dt class="vads-u-display--block vads-u-font-size--h3 vads-u-font-weight--bold vads-u-margin--0">Tinnitus</dt>
  <dd class="vads-u-display--block vads-u-margin--0">
    <dfn class="vads-u-display--inline-block vads-u-font-weight--bold vads-u-margin-y--0 vads-u-margin-right--1">Service-connected disability?</dfn> 
    No</dd>
</dl>
<dl class="vads-u-display--block vads-l-col--12 vads-u-background-color--gray-lightest vads-u-margin-x--0 vads-u-margin-top--0 vads-u-margin-bottom--2 vads-u-padding-top--1 vads-u-padding-bottom--2 vads-u-padding-x--2">
  <dt class="vads-u-display--block vads-u-font-size--h3 vads-u-font-weight--bold vads-u-margin--0"><dfn class="vads-u-display--block vads-u-border-color--gray-light vads-u-border-bottom--1px">10%</dfn>Allergies due to Hearing Loss</dt>
  <dd class="vads-u-display--block vads-u-margin--0">
    <dfn class="vads-u-display--inline-block vads-u-font-weight--bold vads-u-margin-y--0 vads-u-margin-right--1">Service-connected disability?</dfn>
    Yes</dd>
  <dd class="vads-u-display--block vads-u-margin--0">
    <dfn class="vads-u-display--inline-block vads-u-font-weight--bold vads-u-margin-right--0p5">
      Effective date:</dfn>
    01/05/2012</dd>
</dl>
```

### Recommended code example
```html
<dl class="vads-u-display--block vads-l-col--12 vads-u-background-color--gray-lightest vads-u-margin-x--0 vads-u-margin-top--0 vads-u-margin-bottom--2 vads-u-padding-top--1 vads-u-padding-bottom--2 vads-u-padding-x--2">
  <dt class="vads-u-display--block vads-u-font-size--h3 vads-u-font-weight--bold vads-u-margin--0">Tinnitus</dt>
  <dd class="vads-u-display--block vads-u-margin--0">
    <dfn class="vads-u-display--inline-block vads-u-font-weight--bold vads-u-margin-y--0 vads-u-margin-right--1">Service-connected disability?</dfn> 
    No</dd>

  <dt class="vads-u-display--block vads-u-font-size--h3 vads-u-font-weight--bold vads-u-margin--0"><dfn class="vads-u-display--block vads-u-border-color--gray-light vads-u-border-bottom--1px">10%</dfn>Allergies due to Hearing Loss</dt>
  <dd class="vads-u-display--block vads-u-margin--0">
    <dfn class="vads-u-display--inline-block vads-u-font-weight--bold vads-u-margin-y--0 vads-u-margin-right--1">Service-connected disability?</dfn>
    Yes</dd>
  <dd class="vads-u-display--block vads-u-margin--0">
    <dfn class="vads-u-display--inline-block vads-u-font-weight--bold vads-u-margin-right--0p5">
      Effective date:</dfn>
    01/05/2012</dd>
  <!-- insert the remaining items here -->
</dl>
```

**Comment by jenstrickland** [comment]
During testing of NVDA + Firefox, this issue was particularly noticeable. NVDA read each `<dl>` as a list, and said "end of list" for each disability. The list of disabilities would benefit from being a single list.

**Comment by jason-gcio** [comment]
When we circle back into this, for this item it would be a good idea to address the Service-Connected vs not groupings.

**Comment by jenstrickland** [comment]
@jason-gcio Do you have an ETA on when this may be addressed, possibly with the Service-Connect vs Not? I'm sending Rated Disabilities to the VA 508 Office for review.

**Comment by jason-gcio** [comment]
@jenstrickland - We are going to address this when we go back in for iteration and are going to group items like these to reduce the amount of times we open it up.  I don't have an ETA at this time but anticipate it being later in the year, most likely towards the end of the summer.

**Comment by jason-gcio** [comment]
@jenstrickland - can we revisit this when you have a chance?  I think it was mentioned that `<dl>`'s might be bugged (firefox + NVDA) and we would want to use simple `<ol>` and/or `<ul>`'s here.

**Comment by jenstrickland** [comment]
@jason-gcio Yes, Firefox+NVDA states the wrong number of items in the list. So, the new recommended code could be as follows.

- The headings should follow heading order and styling, I just used h# as placeholder.
- Added the `role` attributes because I imagine the design doesn't have bullets, and VoiceOver will need the role to convey the list semantics when `list-style:none` is used to style.

```html

<ol role="list">
  <li role="listitem">
    <h#>Tinnitus</h#>
    <dfn class="vads-u-display--inline-block vads-u-font-weight--bold vads-u-margin-y--0 vads-u-margin-right--1">Service-connected disability?</dfn>  No
  </li>
  <li role="listitem">
    <h#>Allergies due to Hearing Loss</h#>
    <dfn class="vads-u-display--inline-block vads-u-font-weight--bold vads-u-margin-y--0 vads-u-margin-right--1">Service-connected disability?</dfn>  Yes
  </li>
  <li role="listitem">
    <h#>Allergies due to Hearing Loss</h#>
    <ul>
      <li><dfn class="vads-u-display--inline-block vads-u-font-weight--bold vads-u-margin-y--0 vads-u-margin-right--1">Service-connected disability?</dfn> Yes</li>
      <li><dfn class="vads-u-display--inline-block vads-u-font-weight--bold vads-u-margin-right--0p5">
      Effective date:</dfn> 01/05/2012</li>
    </ul>
  </li>
</ol>
  <!-- insert the remaining items here -->
</dl>
```

**Comment by jason-gcio** [comment]
Definition lists might be bugged and work at this time might do more harm than good
